### PR TITLE
Add RPCService unit tests

### LINKED ISSUE #2449: Increase unit tests coverage
### Bug Report
Need  to increase unit tests coverage 

- Add Unit tests to cover changes in Session and Policy Handler 

- Modified Policy Handler Unit tests.

- Add Unit tests for Connection Handler.

- Add Unit tests for RPC Service.

- Add Unit tests for CacheManager.

##### OS & Version Information
* OS/Version: Ubutnu 18
* SDL Core Version: 6.0

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #2449

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
Add RPCService unit tests to improve unit tests coverage (from 59.1% to 95.5% (by functions) and from 27.9% to 69.8% (by lines)).

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gmock/gmock.h>
+#include <string>
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: @ychernysheva reorder includes

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/mock_request.h"
+#include "application_manager/mock_request_controller_settings.h"
+#include "application_manager/mock_rpc_plugin.h"
+#include "application_manager/mock_rpc_plugin_manager.h"
+#include "application_manager/mock_rpc_protection_manager.h"
+#include "application_manager/request_controller.h"
+#include "hmi_message_handler/mock_hmi_message_handler.h"
+#include "include/test/protocol_handler/mock_protocol_handler.h"
+#include "resumption/last_state_impl.h"
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Check that all includes are actually required

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 1 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit 09fa0bdb786a8e4f5634c3e3d121c18a7101c224

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 2 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit 09fa0bdb786a8e4f5634c3e3d121c18a7101c224

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 3]
```diff
@@ -0,0 +1,832 @@
...
+using ::testing::ReturnRef;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+const connection_handler::DeviceHandle kDeviceHandle = 1u;
+const std::string kPolicyAppId = "policy_app_id";
+const uint32_t kCorrelationId = 1u;
+const uint32_t kFunctionId = 1u;
+const uint32_t kAppId = 1u;
+const int32_t hmi_protocol_type_ = 1;
```
Comment (YarikMamykin) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Not used variable - please delete

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 4]
```diff
@@ -0,0 +1,832 @@
...
+const int32_t mobile_protocol_type_ = 0;
```
Comment (YarikMamykin) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Wrong naming, please take a look at constants names above as an example

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 5]
```diff
@@ -0,0 +1,832 @@
...
+  ON_CALL(*mock_app, policy_app_id()).WillByDefault(Return(kPolicyAppId));
+  ON_CALL(mock_app_mngr_, IsAppInReconnectMode(device_id1, kPolicyAppId))
+      .WillByDefault(Return(true));
+
+  EXPECT_CALL(mock_command_holder_,
+              Suspend(static_cast<am::ApplicationSharedPtr>(mock_app),
+                      am::CommandHolder::CommandType::kHmiCommand,
+                      source,
+                      message))
+      .WillOnce(Return());
```
Comment (YarikMamykin) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Did you run check_style script?

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 6]
```diff
@@ -0,0 +1,832 @@
...
+  auto initial_schema_item = initial_schema.getSchemaItem();
+  // Change state of schema and add custom parameter
+  auto new_function_id = mobile_apis::FunctionID::OnButtonEventID;
+  std::map<std::string, SMember> members;
+  members["custom member"] = SMember();
+  rpc_service_with_extended_api->UpdateMobileRPCParams(
+      new_function_id, message_type, members);
+  CSmartSchema updated_schema;
+  extended_mobile_api.GetSchema(new_function_id, message_type, updated_schema);
+  // Check that schema is changed
```
Comment (YarikMamykin) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Place this comment closer to line of code it is related to.
Please check all similar places in file

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 7]
```diff
@@ -0,0 +1,832 @@
...
+  (*message)[am::strings::msg_params][am::strings::request_type] =
+      mobile_apis::RequestType::PROPRIETARY;
+  (*message)[am::strings::msg_params][am::strings::result_code] =
+      mobile_apis::Result::SUCCESS;
+
+  (*message)[jhs::S_PARAMS][jhs::S_FUNCTION_ID] = func_id;
+  (*message)[jhs::S_PARAMS][jhs::S_MESSAGE_TYPE] =
+      mobile_apis::messageType::request;
+  (*message)[jhs::S_PARAMS][jhs::S_PROTOCOL_TYPE] = mobile_protocol_type_;
+  (*message)[jhs::S_PARAMS][jhs::S_PROTOCOL_VERSION] = 1;
```
Comment (YarikMamykin) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Don't use 'magic numbers', use a constant here and declare it in anonymous namespace at the top of file.
Please check all similar places across file.

File: src/components/application_manager/test/rpc_service_impl_test.cc [hunk 8]
```diff
@@ -0,0 +1,832 @@
...
+      hmi_so_factory_,
+      mobile_so_factory_);
+  auto message_to_send = std::make_shared<am::Message>(
+      protocol_handler::MessagePriority::kDefault);
+  message_to_send->set_protocol_version(
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_1);
+  message_to_send->set_connection_key(kConnectionKey);
+  message_to_send->set_correlation_id(kCorrelationId);
+  message_to_send->set_function_id(kFunctionId);
+  message_to_send->set_message_type(static_cast<am::MessageType>(2));
```
Comment (YarikMamykin) [inline src/components/application_manager/test/rpc_service_impl_test.cc]: Magic number. Please provide concrete enum value or make an appropriate constant.

File: tools/infrastructure/collect_coverage.sh [hunk 9]
```diff
@@ -17,7 +17,7 @@ rm -rf $COVERAGE_DIR
 rm -rf $REPORTS_DIR -
 
 mkdir $COVERAGE_DIR
-lcov --quiet --capture --directory . --output-file $COVERAGE_DIR/full_report.info
+lcov --quiet --capture --directory $BUILD_DIR --output-file $COVERAGE_DIR/full_report.info
```
[the remark this answers is not in the document]
Comment (ychernysheva) [reply tools/infrastructure/collect_coverage.sh]: There is no such file in this pull request

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 5 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Yes, and I get such result, I need to edit it manually in addition?

Comment (YarikMamykin) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: @ychernysheva **no**. If script did this then this is the way.

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 3 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 4 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 6 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 7 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 8 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9

File: src/components/application_manager/test/rpc_service_impl_test.cc (hunk 7 above)
Comment (ychernysheva) [reply src/components/application_manager/test/rpc_service_impl_test.cc]: Done in commit d5ec824

**Comment by ychernysheva** [comment]
Rebased on current release branch

**Comment by ychernysheva** [comment]
@theresalech this PR is ready for review. Please notice, that this is **not a regression** issue, so please check it if time allows. Also notice, that this branch was rebased on release/6.1.0 branch, but develop branch is a target for merging.
Thank you.

**Pull request merged**